redis/pool.go

```diff
@@ -299,34 +299,61 @@ func (p *Pool) lazyInit() {
 	p.mu.Unlock()
 }
 
-// get prunes stale connections and returns a connection from the idle list or
-// creates a new connection.
-func (p *Pool) get(ctx context.Context) (*poolConn, error) {
+// waitVacantConn waits while at least one vacant connection appears in the pool.
+// If `ctx` expires before that, an error is returned.
+//
+// waitVacantConn returns:
+// - (as its first return value) the time it spent waiting, if there were no vacant connection in the pool right away
+// or zero otherwise
+// - (as its second return value) error, if any
+func (p *Pool) waitVacantConn(ctx context.Context) (time.Duration, error) {
+	if !p.Wait {
+		return 0, nil
+	}
+	if p.MaxActive <= 0 {
+		return 0, nil
+	}
```

Comment:
These could be combined

Reply:
Actually I've split them "on purpose" for readability reasons, but if you think it reads better when combined I can return it to the original version.

Reply:
yep lets combine and add a comment as per suggestion below.

Reply:
fixed